Add `is_ok` and `is_err` typeguard functions (#135)
rustedpy/result: CHANGELOG.md

## [Unreleased]

> -- `[added]` `is_ok` and `is_err` type guard functions as alternatives to `isinstance` checks (#69)

Reviewer: Note to self, small typo here, remove the extra `-` at the start of this line

> - `[added]` `is_ok` and `is_err` type guard functions. A faster alternative to `isinstance` checks (#69)

Reviewer: Not sure about the wording here. Perhaps we can leave off the `faster`?

How about,
```suggestion
- `[added]` `is_ok` and `is_err` type guard functions as alternatives to `isinstance` (#69)
```
